pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdFunctions.java

```diff
@@ -922,7 +935,7 @@ protected static void startLocalRun(org.apache.pulsar.functions.proto.Function.F
             serviceUrl = DEFAULT_SERVICE_URL;
         }
         try (ProcessRuntimeFactory containerFactory = new ProcessRuntimeFactory(
-                serviceUrl, null, null, null)) {
+                serviceUrl, clientAuthPlugin, clientAuthParam, useTls, allowInsecureConnection, null, null, null)) {
```

Comment:
How abount passing ClientConfigurationData in order to avoid increasing the number of arguments?

Reply:
we can not use `ClientConfigurationData` because it does't have separate fields for plugin-name and params. it also unnecessary creates one auth-provider object which we want to avoid. however, I agree, we need a config class to contain all these params. I will fix it.